### ISSUE TITLE
Reflect functions in the form required by PLE

### ISSUE THREAD

**Issue by facundominguez** [ucsd-progsys/liquidhaskell#1945]
It has been discussed in https://github.com/ucsd-progsys/liquid-fixpoint/pull/517 that functions that have been reflected need to have implications rewritten to `if-then-else`s.

Can we change reflection so the expressions are produced in the form that PLE requires? This would allow simplifying PLE a bit, and probably would make the interface less dependent on the simplifications that PLE can do.

Alternatively, this ticket might amount to documenting the additional constraints that require to leave the code as it is today.

**Comment by nikivazou** [comment]
Yes! Either leave this open or relax the requirement from the fixpoint. I am not sure at the moment that needs to happen for the second, but we can postpone until there is an actual reason to relax it.
